js/compiler/compiler.go

```diff
@@ -119,48 +124,92 @@ func (c *Compiler) Transform(src, filename string) (code string, srcmap []byte,
 		return
 	}
 
-	code, srcmap, err = c.babel.Transform(c.logger, src, filename)
+	code, srcMap, err = c.babel.transformImpl(c.logger, src, filename, c.COpts.SourceMapLoader != nil, inputSrcMap)
 	return
 }
 
+// Options are options to the compiler
+type Options struct {
+	CompatibilityMode lib.CompatibilityMode
+	SourceMapLoader   func(string) ([]byte, error)
+	Strict            bool
+}
+
+// compilationState is helper struct to keep the state of a compilation
+type compilationState struct {
+	// set when we couldn't load external source map so we can try parsing without loading it
+	couldntLoadSourceMap bool
```

Comment:
I'm wondering if we can avoid this property. For example, using the wrapped errors and checks like `errors.Is`. It seems simpler and more go idiomatic, wdyt?

Reply:
The problem @olegbespalov is that the error doesn't get propaged through goja and fixing that will take quite a lot refactoring IIRC, so I opted for *not* trying to do that first.

Reply:
:cry: I see, thanks for clarifying :+1: